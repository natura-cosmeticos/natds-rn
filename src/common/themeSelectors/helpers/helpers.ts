import { Theme, ThemeSelector } from '../theme/theme';

export const buildColorWithOpacity = (
  colorSelector: ThemeSelector,
  opacitySelector: ThemeSelector,
  theme: Theme,
) => {
  const colorHex = colorSelector(theme);
  const opacityScaleOne = opacitySelector(theme);
  const opacityScaleHundred = opacityScaleOne && typeof opacityScaleOne === 'number'
    ? opacityScaleOne * 100
    : '';

  return `${colorHex}${opacityScaleHundred}`;
};
