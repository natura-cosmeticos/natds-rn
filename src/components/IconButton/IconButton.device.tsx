import { storiesOf } from '@storybook/react-native'
import {
  Interactive,
  Default,
  Colors,
  Variants,
  Sizes
} from './IconButton.stories'

storiesOf('IconButton', module)
  .add('Default', Default)
  .add('Variants', Variants)
  .add('Colors', Colors)
  .add('Sizes', Sizes)
  .add('Interactive', Interactive)
