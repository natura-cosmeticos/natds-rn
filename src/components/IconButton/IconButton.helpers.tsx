import { IconColors, IconSizes } from '../Icon/Icon.types'
import {
  getColorByName,
  buildColorWithOpacity,
  getColorHighlight,
  getOpacityVeryHigh,
  getShadowBySize,
  Theme
} from '../../common/themeSelectors'
import {
  IconButtonColors,
  IconContainerProps,
  IconButtonBackgroundStyle,
  IconButtonSizes
} from './IconButton.types'

export const getContainerBackgroundColor = ({
  theme,
  backgroundStyle,
  disabled
}: Omit<IconContainerProps, 'size'> & { theme: Theme }) => {
  const float = disabled
    ? getColorByName(theme, 'lowEmphasis')
    : getColorByName(theme, 'surface')

  const backgroundColor: { [keys in IconButtonBackgroundStyle]: string } = {
    float,
    none: 'transparent',
    overlay: buildColorWithOpacity(getColorHighlight, getOpacityVeryHigh, theme)
  }

  return backgroundColor[backgroundStyle]
}

export const getIconSize = (size: IconButtonSizes) => {
  const iconSize: { [keys in IconButtonSizes]: IconSizes } = {
    medium: 'semiX',
    semi: 'standard',
    semiX: 'semi',
    small: 'standard'
  }

  return iconSize[size]
}

export const getIconColor = (color: IconButtonColors, disabled: boolean) => {
  const highEmphasis = disabled ? 'mediumEmphasis' : 'highEmphasis'

  const iconColor: Record<IconButtonColors, IconColors> = {
    default: highEmphasis,
    highEmphasis,
    light: disabled ? 'lowEmphasis' : 'surface',
    primary: disabled ? 'mediumEmphasis' : 'primary'
  }

  return iconColor[color]
}

export const getContainerElevation = (
  backgroundStyle: IconButtonBackgroundStyle,
  theme: Theme
) => (
  backgroundStyle === 'float'
    ? getShadowBySize(theme, 'micro')
    : {}
)
