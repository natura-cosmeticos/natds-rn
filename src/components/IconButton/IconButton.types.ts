/* eslint-disable max-len */
import { IconName } from '@naturacosmeticos/natds-icons'
import { Size } from '@naturacosmeticos/natds-themes/react-native'
import { ReactElement } from 'react'

export interface IconButtonBaseProps {
  IconComponent: ReactElement;
  backgroundStyle?: IconButtonBackgroundStyle;
  disabled?: boolean;
  onPress?: () => void;
  testID?: string;
  size?: keyof Size;
}

/**
 * @deprecated Use 'semi'
 */
export type IconButtonSizesDeprecated = 'small';
export type IconButtonSizes = 'semi' | 'semiX' | 'medium' | IconButtonSizesDeprecated;

/**
 * @deprecated Use 'highEmphasis'
 */
export type IconButtonColorsDeprecated = 'default';
export type IconButtonColors = 'primary' | 'highEmphasis' | 'light' | IconButtonColorsDeprecated;

export type IconButtonBackgroundStyle = 'none' | 'float' | 'overlay';

export type IconContainerProps = Required<Pick<IconButtonBaseProps, 'size' | 'backgroundStyle' | 'disabled'>>;

export interface IconButtonProps {
  /**
   * An accessibility hint helps users understand what will happen when they perform an action
   * on the accessibility element when that result is not clear from the accessibility label.
   */
  accessibilityHint?: string;
  /**
   * Overrides the text that's read by the screen reader when the user interacts with the element.
   * By default, the label is constructed by traversing all the children and accumulating
   * all the Text nodes separated by space.
   */
  accessibilityLabel?: string;
  /**
   * Icon Button variants `primary` | `highEmphasis` | `light` | `default` - variant `default` is deprecated, use `highEmphasis` instead
   * @default highEmphasis
   */
  iconColor?: IconButtonColors;
  /**
   * Icon Button variants `primary` | `highEmphasis` | `light` | `default` - deprecated use iconColor instead
   * @default highEmphasis
   */
  color?: IconButtonColors;
  /**
   * Icon background style `none` | `float` | `overlay`
   * @default `none`
   */
  backgroundStyle?: IconButtonBackgroundStyle;
  /**
   * Disable user interaction
   * @default false
   */
  disabled?: boolean;
  /**
   * The icon name
   */
  icon?: IconName;
  /**
   * Icon Button sizes `semi` | `semiX` | `medium`;
   * @default `semi`
   * @deprecated size `small`
   */
  size?: IconButtonSizes;
  /**
   * The onPress event handler
   */
  onPress: () => void;
  /**
   * Optional ID for testing
   */
  testID?: string;
}

export const iconButtonSizesOptions: Record<IconButtonSizes, IconButtonSizes> = {
  medium: 'medium',
  semi: 'semiX',
  semiX: 'semiX',
  small: 'small'
}

export const iconButtonColorsOptions: Record<IconButtonColors, IconButtonColors> = {
  default: 'default',
  highEmphasis: 'highEmphasis',
  light: 'light',
  primary: 'primary'
}

export const iconButtonBackgroundStylesOptions: Record<
  IconButtonBackgroundStyle,
  IconButtonBackgroundStyle
> = {
  float: 'float',
  none: 'none',
  overlay: 'overlay'
}

export const iconButtonBackgroundStyles = Object
  .keys(iconButtonBackgroundStylesOptions) as Array<IconButtonBackgroundStyle>
export const iconButtonColors = Object.keys(iconButtonColorsOptions) as Array<IconButtonColors>
export const iconButtonSizes = Object.keys(iconButtonSizesOptions) as Array<IconButtonSizes>
