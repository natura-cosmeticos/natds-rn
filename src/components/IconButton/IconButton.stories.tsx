import React from 'react'
import { select, boolean } from '@storybook/addon-knobs'
import { iconNames, IconName } from '@naturacosmeticos/natds-icons'
import { IconButton } from './IconButton'
import { StoryContainer, StoryWrapper } from '../../common/HelperComponents/StoryContainer'
import {
  iconButtonBackgroundStylesOptions,
  iconButtonSizesOptions,
  iconButtonColorsOptions
} from './IconButton.types'

const extractComponentDescription = () => `
- - -

  ### NOTE:
  This component is available in the following variants:

  - ✅ Standard

With the following attribute status:

  - **Icon Color:**
    - ✅ \`Hight Emphasis\`
    - ✅ \`Primary\`
    - ✅ \`Light\`
  - **Background Style:**
    - ✅ \`None (inherit)\`
    - ✅ \`Float\`
    - ✅ \`Overlay\`
  - ✅ **Disabled**
  - **Size:**
    - ✅ \`Semi\`
    - ✅ \`SemiX\`
    - ✅ \`Medium\`
  - ✅ **OnPress (OnClick)**
  - ✅ **Icon**

- - -
`

export default {
  component: IconButton,
  parameters: {
    componentSubtitle: 'A button component with an icon from @naturacosmeticos/natds-icons package',
    docs: {
      extractComponentDescription
    }
  },
  title: 'Components/IconButton'
}

export const Default = () => (
  <StoryContainer title="Standard">
    <IconButton onPress={() => ({})} />
  </StoryContainer>
)

export const Variants = () => (
  <StoryWrapper title="Variants">
    <StoryContainer title="Standard">
      <IconButton onPress={() => ({})} />
    </StoryContainer>
    <StoryContainer title="Float">
      <IconButton onPress={() => ({})} backgroundStyle="float" />
    </StoryContainer>
    <StoryContainer title="Overlay">
      <IconButton onPress={() => ({})} iconColor="light" backgroundStyle="overlay" />
    </StoryContainer>
    <StoryContainer title="Disabled Standard">
      <IconButton onPress={() => ({})} disabled backgroundStyle="none" />
    </StoryContainer>
    <StoryContainer title="Disabled Float">
      <IconButton onPress={() => ({})} disabled backgroundStyle="float" />
    </StoryContainer>
    <StoryContainer title="Disabled Overlay">
      <IconButton onPress={() => ({})} disabled iconColor="light" backgroundStyle="overlay" />
    </StoryContainer>
  </StoryWrapper>
)

export const Colors = () => (
  <StoryWrapper title="Colors">
    <StoryContainer title="Primary" style={{ backgroundColor: '#eaeaea' }}>
      <IconButton onPress={() => ({})} iconColor="primary" />
    </StoryContainer>
    <StoryContainer title="Light" style={{ backgroundColor: '#eaeaea' }}>
      <IconButton onPress={() => ({})} iconColor="light" />
    </StoryContainer>
    <StoryContainer title="HighEmphasis" style={{ backgroundColor: '#eaeaea' }}>
      <IconButton onPress={() => ({})} iconColor="highEmphasis" />
    </StoryContainer>
  </StoryWrapper>
)

export const Sizes = () => (
  <StoryWrapper title="Sizes">
    <StoryContainer title="Semi">
      <IconButton onPress={() => ({})} size="semi" />
    </StoryContainer>
    <StoryContainer title="SemiX">
      <IconButton onPress={() => ({})} size="semiX" />
    </StoryContainer>
    <StoryContainer title="Medium">
      <IconButton onPress={() => ({})} size="medium" />
    </StoryContainer>
  </StoryWrapper>
)

export const Interactive = () => (
  <StoryContainer title="Interactive">
    <IconButton
      iconColor={select('Colors', iconButtonColorsOptions, 'default')}
      icon={select('Icon name', iconNames as Array<IconName>, 'outlined-finance-bank' as IconName)}
      size={select('Sizes', iconButtonSizesOptions, 'small')}
      disabled={boolean('Disabled', false)}
      onPress={() => ({})}
      backgroundStyle={select('BackgroundStyle', iconButtonBackgroundStylesOptions, iconButtonBackgroundStylesOptions.none)}
    />
  </StoryContainer>
)
