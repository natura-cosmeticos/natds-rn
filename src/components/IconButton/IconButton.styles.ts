import styled from 'styled-components/native'
import { css } from 'styled-components'
import {
  getContainerBackgroundColor,
  getContainerElevation
} from './IconButton.helpers'
import { getSize } from '../../common/themeSelectors'

import { IconContainerProps } from './IconButton.types'

export const IconContainer = styled.View<IconContainerProps>`
  ${({
    theme, backgroundStyle, disabled, size
  }) => css`
    ${getContainerElevation(backgroundStyle, theme)};
    height: ${getSize(theme, size)}px;
    width: ${getSize(theme, size)}px;
    background-color: ${getContainerBackgroundColor({ backgroundStyle, disabled, theme })};
    border-radius: 50px;
    justify-content: center;
    align-items: center;
  `}
`
