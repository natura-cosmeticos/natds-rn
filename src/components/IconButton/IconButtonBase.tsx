import React from 'react';
import { useTheme } from 'styled-components/native';
import { TouchableRipple } from '../TouchableRipple/TouchableRipple';
import { getSize } from '../../common/themeSelectors';
import {
  getContainerElevation,
} from './IconButton.helpers';
import { IconButtonBaseProps } from './IconButton.types';
import { IconContainer } from './IconButton.styles';

export const IconButtonBase = ({
  size = 'semi',
  IconComponent,
  backgroundStyle = 'none',
  disabled = false,
  onPress,
  testID,
}: IconButtonBaseProps) => {
  const theme = useTheme();

  return (
    <TouchableRipple
      color="highlight"
      size={getSize(theme, size) / 2 + 5}
      onPress={onPress}
      disabled={disabled}
      testID={`${testID}-button`}
      style={{ borderRadius: 50 }}
    >
      <IconContainer
        disabled={disabled}
        size={size}
        backgroundStyle={backgroundStyle}
        style={getContainerElevation(backgroundStyle, theme)}
        testID={`${testID}-background`}
      >
       {IconComponent}
      </IconContainer>
    </TouchableRipple>
  );
};
