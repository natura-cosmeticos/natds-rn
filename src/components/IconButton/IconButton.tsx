import React from 'react'
import { Icon } from '../Icon/Icon'
import { IconButtonProps } from './IconButton.types'
import {
  getIconColor,
  getIconSize
} from './IconButton.helpers'
import { IconButtonBase } from './IconButtonBase'

export const IconButton = ({
  accessibilityHint,
  accessibilityLabel,
  color = 'highEmphasis',
  iconColor = color,
  icon,
  size = 'semi',
  backgroundStyle = 'none',
  disabled = false,
  onPress,
  testID = 'ds-icon-button'
}: IconButtonProps) => (
  <IconButtonBase
    disabled={disabled}
    size={size}
    backgroundStyle={backgroundStyle}
    testID={testID}
    onPress={disabled ? undefined : onPress}
    IconComponent={(
      <Icon
        accessibilityHint={accessibilityHint}
        accessibilityLabel={accessibilityLabel}
        accessibilityRole="imagebutton"
        size={getIconSize(size)}
        color={getIconColor(iconColor, disabled)}
        name={icon}
        testID={`${testID}-icon`}
      />
      )}
  />
)
