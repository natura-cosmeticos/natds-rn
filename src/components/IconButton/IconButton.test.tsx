/* eslint-disable max-statements, mocha/no-setup-in-describe */
import React from 'react';
import { fireEvent } from '@testing-library/react-native';
import mockTheme from '../../common/themeSelectors/theme/mock-theme.json';
import { renderWithTheme } from '../../../test/testHelpers';
import { IconButton } from './IconButton';
import * as helpers from './IconButton.helpers';
import { iconButtonSizes, iconButtonColors, iconButtonBackgroundStyles } from './IconButton.types';

jest.mock('../Icon/Icon');
jest.mock('../../common/themeSelectors');

describe('IconButton component', () => {
  let getContainerElevationMock: jest.SpyInstance;
  let getContainerBackgroundColorMock: jest.SpyInstance;
  let getIconColorMock: jest.SpyInstance;
  let getIconSizeMock: jest.SpyInstance;

  afterEach(() => {
    jest.restoreAllMocks();
  });

  beforeEach(() => {
    // @ts-ignore
    getContainerElevationMock = jest
      .spyOn(helpers, 'getContainerElevation')
      .mockReturnValue({ elevation: 1 });
    // @ts-ignore
    getIconColorMock = jest
      .spyOn(helpers, 'getIconColor')
      .mockReturnValue('highEmphasis');
    // @ts-ignore
    getIconSizeMock = jest
      .spyOn(helpers, 'getIconSize')
      .mockReturnValue('standard');
    // @ts-ignore
    getContainerBackgroundColorMock = jest
      .spyOn(helpers, 'getContainerBackgroundColor')
      .mockReturnValue('transparent');
  });

  it('should render the correct background', () => {
    renderWithTheme(
      <IconButton
        iconColor="default"
        icon="outlined-finance-bank"
        size="semiX"
        backgroundStyle="float"
        onPress={() => { }}
      />,
    );

    expect(getContainerElevationMock).toHaveBeenCalledWith('float', mockTheme);
    expect(getContainerBackgroundColorMock).toHaveBeenCalledWith({
      backgroundStyle: 'float',
      disabled: false,
      theme: mockTheme,
    });
  });

  it('should render the correct icon', () => {
    const { queryByTestId } = renderWithTheme(
      <IconButton
        iconColor="primary"
        icon="outlined-finance-bank"
        accessibilityHint="Go to account page"
        accessibilityLabel="Press view your account"
        size="semiX"
        backgroundStyle="float"
        onPress={() => { }}
      />,
    );

    const icon = queryByTestId('icon-button-icon');

    expect(icon?.props).toHaveProperty('accessibilityHint', 'Go to account page');
    expect(icon?.props).toHaveProperty('accessibilityLabel', 'Press view your account');
    expect(icon?.props).toHaveProperty('size', 'standard');
    expect(icon?.props).toHaveProperty('color', 'highEmphasis');
    expect(icon?.props).toHaveProperty('name', 'outlined-finance-bank');
    expect(getIconSizeMock).toHaveBeenCalledWith('semiX');
    expect(getIconColorMock).toHaveBeenCalledWith('primary', false);
  });
  it('should call onPress when item is pressed', () => {
    const onPressMock = jest.fn();

    const { queryByTestId } = renderWithTheme(
      <IconButton
        color="default"
        icon="outlined-finance-bank"
        onPress={onPressMock}
      />,
    );

    const iconButton = queryByTestId('icon-button');

    if (iconButton) {
      fireEvent.press(iconButton);
    }

    expect(onPressMock).toHaveBeenCalled();
  });

  it.each(iconButtonSizes)('should render IconButton component with size %p', (size) => {
    const { toJSON } = renderWithTheme(
      <IconButton
        size={size}
        accessibilityHint="Go to account page"
        accessibilityLabel="Press view your account"
        icon="outlined-finance-bank"
        onPress={() => { }}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it.each(iconButtonBackgroundStyles)('should render IconButton component with background style %p', (backgroundStyle) => {
    const { toJSON } = renderWithTheme(
      <IconButton
        backgroundStyle={backgroundStyle}
        accessibilityHint="Go to account page"
        accessibilityLabel="Press view your account"
        icon="outlined-finance-bank"
        onPress={() => { }}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it.each(iconButtonColors)('should render IconButton component with color %p', (iconColor) => {
    const { toJSON } = renderWithTheme(
      <IconButton
        iconColor={iconColor}
        accessibilityHint="Go to account page"
        accessibilityLabel="Press view your account"
        icon="outlined-finance-bank"
        onPress={() => { }}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render IconButton component disabled', () => {
    const { toJSON } = renderWithTheme(
      <IconButton
        disabled
        accessibilityHint="Go to account page"
        accessibilityLabel="Press view your account"
        icon="outlined-finance-bank"
        onPress={() => { }}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
