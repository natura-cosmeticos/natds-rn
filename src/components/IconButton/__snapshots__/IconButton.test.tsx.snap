// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton component should render IconButton component disabled 1`] = `
<View
  accessible={true}
  color="highlight"
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size={15}
  testID="icon-button"
>
  <View
    backgroundStyle="none"
    disabled={true}
    size="semi"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderRadius": 50,
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        },
        Object {
          "elevation": 1,
        },
      ]
    }
    testID="icon-button-background"
  >
    <Text
      accessibilityHint="Go to account page"
      accessibilityLabel="Press view your account"
      accessibilityRole="imagebutton"
      color="highEmphasis"
      name="outlined-finance-bank"
      size="standard"
      testID="icon-button-icon"
    >
      outlined-finance-bank
    </Text>
  </View>
</View>
`;

exports[`IconButton component should render IconButton component with background style "float" 1`] = `
<View
  accessible={true}
  color="highlight"
  focusable={true}
  onClick={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size={15}
  testID="icon-button"
>
  <View
    backgroundStyle="float"
    disabled={false}
    size="semi"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderRadius": 50,
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        },
        Object {
          "elevation": 1,
        },
      ]
    }
    testID="icon-button-background"
  >
    <Text
      accessibilityHint="Go to account page"
      accessibilityLabel="Press view your account"
      accessibilityRole="imagebutton"
      color="highEmphasis"
      name="outlined-finance-bank"
      size="standard"
      testID="icon-button-icon"
    >
      outlined-finance-bank
    </Text>
  </View>
</View>
`;

exports[`IconButton component should render IconButton component with background style "none" 1`] = `
<View
  accessible={true}
  color="highlight"
  focusable={true}
  onClick={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size={15}
  testID="icon-button"
>
  <View
    backgroundStyle="none"
    disabled={false}
    size="semi"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderRadius": 50,
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        },
        Object {
          "elevation": 1,
        },
      ]
    }
    testID="icon-button-background"
  >
    <Text
      accessibilityHint="Go to account page"
      accessibilityLabel="Press view your account"
      accessibilityRole="imagebutton"
      color="highEmphasis"
      name="outlined-finance-bank"
      size="standard"
      testID="icon-button-icon"
    >
      outlined-finance-bank
    </Text>
  </View>
</View>
`;

exports[`IconButton component should render IconButton component with background style "overlay" 1`] = `
<View
  accessible={true}
  color="highlight"
  focusable={true}
  onClick={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size={15}
  testID="icon-button"
>
  <View
    backgroundStyle="overlay"
    disabled={false}
    size="semi"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderRadius": 50,
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        },
        Object {
          "elevation": 1,
        },
      ]
    }
    testID="icon-button-background"
  >
    <Text
      accessibilityHint="Go to account page"
      accessibilityLabel="Press view your account"
      accessibilityRole="imagebutton"
      color="highEmphasis"
      name="outlined-finance-bank"
      size="standard"
      testID="icon-button-icon"
    >
      outlined-finance-bank
    </Text>
  </View>
</View>
`;

exports[`IconButton component should render IconButton component with color "default" 1`] = `
<View
  accessible={true}
  color="highlight"
  focusable={true}
  onClick={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size={15}
  testID="icon-button"
>
  <View
    backgroundStyle="none"
    disabled={false}
    size="semi"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderRadius": 50,
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        },
        Object {
          "elevation": 1,
        },
      ]
    }
    testID="icon-button-background"
  >
    <Text
      accessibilityHint="Go to account page"
      accessibilityLabel="Press view your account"
      accessibilityRole="imagebutton"
      color="highEmphasis"
      name="outlined-finance-bank"
      size="standard"
      testID="icon-button-icon"
    >
      outlined-finance-bank
    </Text>
  </View>
</View>
`;

exports[`IconButton component should render IconButton component with color "highEmphasis" 1`] = `
<View
  accessible={true}
  color="highlight"
  focusable={true}
  onClick={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size={15}
  testID="icon-button"
>
  <View
    backgroundStyle="none"
    disabled={false}
    size="semi"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderRadius": 50,
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        },
        Object {
          "elevation": 1,
        },
      ]
    }
    testID="icon-button-background"
  >
    <Text
      accessibilityHint="Go to account page"
      accessibilityLabel="Press view your account"
      accessibilityRole="imagebutton"
      color="highEmphasis"
      name="outlined-finance-bank"
      size="standard"
      testID="icon-button-icon"
    >
      outlined-finance-bank
    </Text>
  </View>
</View>
`;

exports[`IconButton component should render IconButton component with color "light" 1`] = `
<View
  accessible={true}
  color="highlight"
  focusable={true}
  onClick={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size={15}
  testID="icon-button"
>
  <View
    backgroundStyle="none"
    disabled={false}
    size="semi"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderRadius": 50,
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        },
        Object {
          "elevation": 1,
        },
      ]
    }
    testID="icon-button-background"
  >
    <Text
      accessibilityHint="Go to account page"
      accessibilityLabel="Press view your account"
      accessibilityRole="imagebutton"
      color="highEmphasis"
      name="outlined-finance-bank"
      size="standard"
      testID="icon-button-icon"
    >
      outlined-finance-bank
    </Text>
  </View>
</View>
`;

exports[`IconButton component should render IconButton component with color "primary" 1`] = `
<View
  accessible={true}
  color="highlight"
  focusable={true}
  onClick={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size={15}
  testID="icon-button"
>
  <View
    backgroundStyle="none"
    disabled={false}
    size="semi"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderRadius": 50,
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        },
        Object {
          "elevation": 1,
        },
      ]
    }
    testID="icon-button-background"
  >
    <Text
      accessibilityHint="Go to account page"
      accessibilityLabel="Press view your account"
      accessibilityRole="imagebutton"
      color="highEmphasis"
      name="outlined-finance-bank"
      size="standard"
      testID="icon-button-icon"
    >
      outlined-finance-bank
    </Text>
  </View>
</View>
`;

exports[`IconButton component should render IconButton component with size "medium" 1`] = `
<View
  accessible={true}
  color="highlight"
  focusable={true}
  onClick={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size={15}
  testID="icon-button"
>
  <View
    backgroundStyle="none"
    disabled={false}
    size="medium"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderRadius": 50,
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        },
        Object {
          "elevation": 1,
        },
      ]
    }
    testID="icon-button-background"
  >
    <Text
      accessibilityHint="Go to account page"
      accessibilityLabel="Press view your account"
      accessibilityRole="imagebutton"
      color="highEmphasis"
      name="outlined-finance-bank"
      size="standard"
      testID="icon-button-icon"
    >
      outlined-finance-bank
    </Text>
  </View>
</View>
`;

exports[`IconButton component should render IconButton component with size "semi" 1`] = `
<View
  accessible={true}
  color="highlight"
  focusable={true}
  onClick={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size={15}
  testID="icon-button"
>
  <View
    backgroundStyle="none"
    disabled={false}
    size="semi"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderRadius": 50,
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        },
        Object {
          "elevation": 1,
        },
      ]
    }
    testID="icon-button-background"
  >
    <Text
      accessibilityHint="Go to account page"
      accessibilityLabel="Press view your account"
      accessibilityRole="imagebutton"
      color="highEmphasis"
      name="outlined-finance-bank"
      size="standard"
      testID="icon-button-icon"
    >
      outlined-finance-bank
    </Text>
  </View>
</View>
`;

exports[`IconButton component should render IconButton component with size "semiX" 1`] = `
<View
  accessible={true}
  color="highlight"
  focusable={true}
  onClick={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size={15}
  testID="icon-button"
>
  <View
    backgroundStyle="none"
    disabled={false}
    size="semiX"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderRadius": 50,
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        },
        Object {
          "elevation": 1,
        },
      ]
    }
    testID="icon-button-background"
  >
    <Text
      accessibilityHint="Go to account page"
      accessibilityLabel="Press view your account"
      accessibilityRole="imagebutton"
      color="highEmphasis"
      name="outlined-finance-bank"
      size="standard"
      testID="icon-button-icon"
    >
      outlined-finance-bank
    </Text>
  </View>
</View>
`;

exports[`IconButton component should render IconButton component with size "small" 1`] = `
<View
  accessible={true}
  color="highlight"
  focusable={true}
  onClick={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size={15}
  testID="icon-button"
>
  <View
    backgroundStyle="none"
    disabled={false}
    size="small"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderRadius": 50,
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        },
        Object {
          "elevation": 1,
        },
      ]
    }
    testID="icon-button-background"
  >
    <Text
      accessibilityHint="Go to account page"
      accessibilityLabel="Press view your account"
      accessibilityRole="imagebutton"
      color="highEmphasis"
      name="outlined-finance-bank"
      size="standard"
      testID="icon-button-icon"
    >
      outlined-finance-bank
    </Text>
  </View>
</View>
`;
